controllers/cluster-registration/cluster_registration_controller.go

```diff
@@ -29,27 +50,216 @@ type RegisteredClusterReconciler struct {
 	APIExtensionClient apiextensionsclient.Interface
 	Log                logr.Logger
 	Scheme             *runtime.Scheme
+	MceCluster         []helpers.MceInstance
 }
 
 func (r *RegisteredClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = context.Background()
 	logger := r.Log.WithValues("namespace", req.Namespace, "name", req.Name)
 	logger.Info("Reconciling...")
 
+	instance := &singaporev1alpha1.RegisteredCluster{}
+
+	if err := r.Client.Get(
+		context.TODO(),
+		types.NamespacedName{Namespace: req.Namespace, Name: req.Name},
+		instance,
+	); err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, giterrors.WithStack(err)
+	}
+
+	// create managecluster on creation of registeredcluster CR
+	if err := r.createManagedCluster(instance, ctx); err != nil {
+		logger.Error(err, "failed to create ManagedCluster")
+
+		return ctrl.Result{}, err
+	}
+
+	// update status of registeredcluster - add import command
+	if err := r.updateImportCommand(instance, ctx); err != nil {
+		logger.Error(err, "failed to update import command")
+		if errors.IsNotFound(err) {
+			return reconcile.Result{Requeue: true, RequeueAfter: 1 * time.Second}, nil
```

Comment:
@itdove I have added requeue after 1 sec if import secret is not found. But the conatiner logs shows that error. is there any way we can avoid showing that error in logs.

Reply:
You mean the error for line 87? Let's just move it down after the if errors.IsNotFound check.